packages/api/endpoints/executions.js

```diff
@@ -79,6 +85,29 @@ async function del(req, res) {
   return res.send({ message: 'Record deleted' });
 }
 
+/**
+ * Get execution history for a single granule or multiple granules
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function searchByGranules(req, res) {
+  const payload = req.body;
+  const knex = await getKnexClient();
+  const granules = await getGranulesForPayload(payload, knex);
+
+  const executionPgModel = new ExecutionPgModel();
+
+  const executionCumulusIds = await getApiGranuleExecutionCumulusIds(knex, granules);
+
+  const executions = await executionPgModel
+    .searchByCumulusIds(knex, executionCumulusIds);
+
+  return res.send(executions);
```

Comment:
It just occurs to me that we're returning execution records directly out of the PG database here. While there is nothing wrong with that per se, I thought the intention for our current API work was to still return records in the "API" format (basically matching the existing structure of Dynamo records to preserve backwards compatibility with our API). I guess that would be up to @laurenfrederick 

If you do need to do that, then you can take this `translatePostgresExecutionToApiExecution` helper straight out of the feature branch and run it for each execution being returned:

https://github.com/nasa/cumulus/blob/feature/rds-phase-2/packages/db/src/translate/executions.ts#L13

That being said, it also occurs to me that especially when using an ES query to return a bunch of results, there could be many executions returned, so should this endpoint return a limited set of results by default and support paging? Otherwise for a really broad granule query the lambda would probably run out of memory and crash

Reply:
Yeah you're absolutely right - the records should be in the API format. Grabbed that function from the feature branch.

I added some paging functionality to `executionPgModel.searchByCumulusIds` and some corresponding tests. I set the default limit and page to 1 in order to match the other Cumulus API endpoints. Hopefully that's sufficient